new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/LoggingUtil.java

```diff
@@ -137,4 +232,162 @@ private static void writeToFile(Object obj, String resultDir, String fileName) t
     }
   }
 
-}
+  /**
+   * Copy files from persistent volume to local folder.
+   * @param namespace name of the namespace
+   * @param hostPath the persistent volume host path
+   * @param destinationPath destination folder to copy the files to
+   * @throws ApiException when pod interaction fails
+   */
+  private static void copyFromPV(String namespace, String hostPath, Path destinationPath) throws ApiException {
+    V1Pod pvPod = null;
+    try {
+      // create a temporary pod to get access to the interested persistent volume
+      pvPod = setupPVPod(namespace, hostPath);
+      copyDirectoryFromPod(pvPod, hostPath, destinationPath);
+    } finally {
+      // remove the temporary pod
+      if (pvPod != null) {
+        cleanupPVPod(namespace);
+      }
+    }
+  }
+
+  /**
+   * Creates temporary pod with persistent volume claim and persistent volume using host path.
+   *
+   * @param namespace name of the namespace
+   * @param hostPath host path from ineterested persistent volume
+   * @return V1Pod pod object
+   * @throws ApiException when create pod fails
+   */
+  private static V1Pod setupPVPod(String namespace, String hostPath) throws ApiException {
+
+    ConditionFactory withStandardRetryPolicy = with().pollDelay(2, SECONDS)
+        .and().with().pollInterval(5, SECONDS)
+        .atMost(1, MINUTES).await();
+
+    // create a pvc and pv to get access to the host path of the target pv
+    final String pvcName = "pv-pod-pvc-" + namespace;
+    final String pvName = "pv-pod-pv-" + namespace;
+
+    V1PersistentVolume v1pv = new V1PersistentVolume()
```

Comment:
i am lost here - why are you creating a PV?

Reply:
i don't understand what this method is doing. javadoc says it is copying from a pv, so the pv must already exist, right?

Reply:
The initial implementation was to create a new one-off pod with its own pvc and pv (using the hostPath location from the pv used by the test, where the contents need to be archived are located)

Now I changed the code such that the one-off pod will use the existing pvc/pv used by the test. Mount it under /shared in one-off pod and copy it.